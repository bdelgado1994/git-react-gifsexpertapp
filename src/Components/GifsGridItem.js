import React from 'react';
import PropTypes from 'prop-types';


export default function GifsGridItem({ title, url }) {
    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}
GifsGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}