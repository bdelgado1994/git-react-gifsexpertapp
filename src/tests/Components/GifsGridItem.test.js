import React from 'react';
import { shallow } from 'enzyme';
import GifsGridItem from '../../Components/GifsGridItem';

describe('Prueba Unitaria del componente GifsGridItem', () => {
    const titulo = 'Un titulo';
    const url = 'localhost/prueba.jpg';
    const wrapper = shallow(<GifsGridItem
        title={titulo}
        url={url}
    />)
    test('Prueba del componente', () => {

        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de Tener Un parrafo con el title', () => {
        const textoTitulo = wrapper.find('p')
        expect(textoTitulo.text().trim()).toBe(titulo);
    })
    test('debe de tner la imagen igual url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log(img.prop('src'))
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo);
    });
    test('el div debe de tener la clase animate__bounce', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('animate__bounce')).toBe(true)
    })


})
