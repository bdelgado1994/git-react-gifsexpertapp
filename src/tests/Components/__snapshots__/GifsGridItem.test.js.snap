// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Unitaria del componente GifsGridItem Prueba del componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="localhost/prueba.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
